staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -102,7 +103,8 @@ func NewSchemaValidator(customResourceValidation *apiextensions.JSONSchemaProps)
 	openapiSchema := &spec.Schema{}
 	if customResourceValidation != nil {
 		// TODO: replace with NewStructural(...).ToGoOpenAPI
-		if err := ConvertJSONSchemaPropsWithPostProcess(customResourceValidation, openapiSchema, StripUnsupportedFormatsPostProcess); err != nil {
+		formatPostProcessor := StripUnsupportedFormatsPostProcessorForVersion(environment.DefaultCompatibilityVersion())
```

Comment:
@siyuanfoundation @Jefftree We should really move `environment.DefaultCompatibilityVersion()` outside of the CEL libraries.  It's generally useful.  I'll leave that for a follow up PR.

Reply:
+1